### DISCUSSION TITLE
Export monai code to bundle (json/yml file)?

### DISCUSSION THREAD

**Discussion by dyollb** [Project-MONAI/MONAI#4535] [Q&A]
I am looking into using the bundle feature to parametrize the transforms.
The example [here](https://github.com/Project-MONAI/MONAI/discussions/4237) show how to load the config.

Is there any way to export the config from a monai code (or less ambitious an existing transform list)?

**Comment by Nic-Ma** [comment]
Hi @dyollb ,

Thanks for your interest in the MONAI and bundle.
We can't convert python code to config so far, because as you know, the python code is not structured content, it's not easy to parse a python file and convert. Actually, structured config is one of the most important features that other applications can easily reconstruct the program.

Thanks.

**Comment by dyollb** [comment]
Hi @Nic-Ma 

I guess for my use case I could export the transform chain via some introspection, e.g. 
- get member variables via `vars(tr)`
- collect (keep only) member variables that have same name in `__init__` method
- store this information, together with class name (`SomeTransform.__init__`) as json/yml

Generalizing this to unstructured code without adding any user-code is impossible, But I guess the user could manually call `monai.bunlde.dump(optimizer)`, `monai.bunlde.dump(pre_processing_transforms)`, etc.

And lightning code could automate this, since it is quite structured..

Anyway, if I implement something like the above I will post it here.

Thanks
